FBTweak/FBTweak.h

```diff
@@ -37,132 +37,167 @@ typedef id FBTweakValue;
-    possible values when changed. Must not be set on actions.
+ @abstract The current value of the tweak.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue currentValue;
+@property (readwrite, nonatomic, nullable) FBTweakValue currentValue;
 
 /**
-  @abstract The possible values of the tweak.
-  @discussion Optional. If nil, any value is allowed. If an
-    FBTweakNumericRange, represents a range of numeric values.
-    If an array or dictionary, contains all of the allowed values.
-    Should not be set on tweaks representing actions.
+ @abstract The default value of the tweak. It is up to the user whether to use this value when the
+ \c currentValue is \c nil.
  */
-@property (nonatomic, strong, readwrite) id possibleValues;
+@property (readonly ,nonatomic) FBTweakValue defaultValue;
 
 /**
-  @abstract The minimum value of the tweak.
-  @discussion Optional. If nil, there is no minimum. Numeric only.
-    Should not be set on tweaks representing actions.
+ @abstract Reset the tweak to its initial state.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue minimumValue;
+- (void)reset;
 
 /**
-  @abstract The maximum value of the tweak.
-  @discussion Optional. If nil, there is no maximum. Numeric only.
-    Should not be set on tweaks representing actions.
+ @abstract The possible values of the tweak.
+ @discussion If nil, any value is allowed. If an FBTweakNumericRange, represents a range of numeric
+ values. If \c NSSArray or \c NSDictionary, contains all of the allowed values.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue maximumValue;
+@property (readwrite, nonatomic, nullable) id possibleValues;
 
 /**
-  @abstract The step value of the tweak.
-  @discussion Optional. If nil, the step value is calculated from
-    the miniumum and maxium values. Only used for numeric tweaks.
+ @abstract The minimum value of the tweak.
+ @discussion If \c nil, there is no minimum. Applicable only when \c currentValue is expected to
+ contain numeric values.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue stepValue;
+@property (readwrite, nonatomic, nullable) FBTweakValue minimumValue;
 
 /**
-  @abstract The decimal precision value of the tweak.
-  @discussion Optional. If nil, the precision value is calculated from
-    the step value. Only used for numeric tweaks.
+ @abstract The maximum value of the tweak.
+ @discussion If \c nil, there is no maximum. Applicable only when \c currentValue is expected to
+ contain numeric values.
  */
-@property (nonatomic, strong, readwrite) FBTweakValue precisionValue;
+@property (readwrite, nonatomic, nullable) FBTweakValue maximumValue;
 
 /**
-  @abstract Adds an observer to the tweak.
-  @param observer The observer. Must not be nil.
-  @discussion A weak reference is taken on the observer.
+ @abstract The step value of the tweak.
+ @discussion If nil, the step value is calculated from the minimum and maximum values. Applicable
```

Comment:
* If this the `stepValue` calculation method isn't defined here, than `step value is calculated from the minimum and maximum values` should be omitted. Cause I can defined a constant `stepValue` which doesn't relay on `min/maxValue`s. Rather and an explanation what it does.
Something like `The value by which the \c currentValue can be changed.`
* Same everywhere else.

Reply:
I wrote, `If \c nil, a default value will be used`. Sounds good?